AdyenActions/Components/3DS2/Action handlers/ThreeDS2CoreActionHandler.swift

```diff
@@ -144,24 +183,66 @@ internal class ThreeDS2CoreActionHandler: Component {
         }
 
     }
+    
+    /// Invoked to handle the error flow of a challenge handling by the 3ds2sdk.
+    /// For challenge cancelled we return the control back to the merchant immediately as an error.
+    private func didReceiveErrorOnChallenge(error: Error?,
+                                            challengeAction: ThreeDS2ChallengeAction,
+                                            completionHandler: @escaping (Result<ThreeDSResult, ThreeDS2CoreActionHandlerError>) -> Void) {
+        guard let error = error as? NSError else {
+            didFail(with: .unknown(UnknownError(errorDescription: "Both error and result are nil, this should never happen.")),
+                    completionHandler: completionHandler)
+            return
+        }
+        switch (error.domain, error.code) {
+        case (ADYRuntimeErrorDomain, Int(ADYRuntimeErrorCode.challengeCancelled.rawValue)):
+            do {
+                let cancellationResult = try ThreeDSResult(authorizationToken: challengeAction.authorisationToken,
+                                                           threeDS2SDKError: error.base64Representation())
+                let cancellationError = ThreeDS2CoreActionHandlerError.cancellationAction(cancellationResult)
+                didFail(with: cancellationError,
+                        completionHandler: completionHandler)
+            } catch {
+                didFail(with: .unknown(UnknownError(errorDescription: "Unable to create ThreeDSResult on error.")),
```

Comment:
do the merchant still receives `UnknownError` ? or `.unknown(UnknownError(errorDescription: "Unable to create ThreeDSResult on error."))` because this would be a breaking change :-)